lms/djangoapps/experiments/flags.py

```diff
@@ -158,17 +181,17 @@ def get_bucket(self, course_key=None, track=True):
                 break
 
         session_key = 'tracked.{}'.format(experiment_name)
-        if track and hasattr(request, 'session') and session_key not in request.session:
+        if track and hasattr(request, 'session') and session_key not in request.session and user == request.user:
```

Comment:
If the code above this overwrote `request.user`, this won't work. You really just want to check the value of `is_masquerading_as_specific_student` probably?